apps/api/src/app/workflows/dto/create-notification-template.request.dto.ts

```diff
@@ -0,0 +1,65 @@
+import { IsArray, IsBoolean, IsDefined, IsOptional, IsString, MaxLength, ValidateNested } from 'class-validator';
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { ICreateNotificationTemplateDto, IPreferenceChannels } from '@novu/shared';
+
+import { PreferenceChannels } from '../../shared/dtos/preference-channels';
+import { NotificationStep } from '../../shared/dtos/notification-step';
+
+export class CreateWorkflowRequestDto implements ICreateNotificationTemplateDto {
```

Comment:
```suggestion
export class CreateWorkflowRequestDto implements ICreateWorkflowDto {
```
We would need to rename the implemented interface. You might need to duplicate it to not affect the deprecated API or just change the implemented DTO also for the deprecated API.